hyperspy/signal.py

```diff
@@ -3213,20 +3218,27 @@ def fft(self, shifted=False, **kwargs):
 
         if self.axes_manager.signal_dimension == 0:
             raise AttributeError("Signal dimension must be at least one.")
+        if apodization==True:
+            apodization = 'hann'
+
+        if apodization:
+            im_fft = self.deepcopy().apply_apodization(type=apodization)
+        else:
+            im_fft = self.deepcopy()
```

Comment:
`im_fft = self` - no need to make an extra copy of a (potentially) large signal into memory here. It's already copied further down. The apodization=True case is correct though.